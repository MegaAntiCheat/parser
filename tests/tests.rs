use std::fs;
use test_case::test_case;

use tf_demo_parser::demo::parser::gamestateanalyser::GameStateAnalyser;
use tf_demo_parser::{Demo, DemoParser};

#[test_case("small.dem")]
#[test_case("gully.dem")]
#[test_case("comp.dem")]
#[test_case("malformed_cvar.dem")]
#[test_case("unicode-saytext.dem")]
#[test_case("nousers.dem")]
#[test_case("decal.dem")]
#[test_case("saytext2.dem")]
#[test_case("emptysaytext.dem")]
#[test_case("protocol23.dem")]
fn snapshot_test(input_file: &str) {
    let file = fs::read(format!("test_data/{}", input_file)).expect("Unable to read file");
    let demo = Demo::new(&file);
    let (_, state) = DemoParser::new(demo.get_stream()).parse().unwrap();

    insta::assert_json_snapshot!(format!("{input_file}_minimal"), state);

    let (_, state) = DemoParser::new_all(demo.get_stream()).parse().unwrap();
    insta::assert_json_snapshot!(format!("{input_file}_minimal"), state);
}

#[test_case("small.dem")]
#[test_case("gully.dem")]
fn game_state_test(input_file: &str) {
    let file = fs::read(format!("test_data/{}", input_file)).expect("Unable to read file");
    let demo = Demo::new(&file);
    let (_, state) = DemoParser::new_with_analyser(demo.get_stream(), GameStateAnalyser::new())
        .parse()
        .unwrap();

    insta::assert_json_snapshot!(input_file, state);
}
