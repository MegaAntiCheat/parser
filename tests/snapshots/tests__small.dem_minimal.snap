---
source: tests/tests.rs
expression: state
---
{
  "chat": [
    {
      "kind": "Empty",
      "from": "",
      "text": "#TF_timeleft",
      "tick": 4
    }
  ],
  "users": {
    "2": {
      "classes": {},
      "name": "Icewind | demos.tf",
      "userId": 2,
      "steamId": "[U:1:64229260]",
      "team": "other"
    }
  },
  "deaths": [],
  "rounds": [],
  "startTick": 68,
  "intervalPerTick": 0.015,
  "pauses": []
}
